### PR TITLE
WIP Kylepres/haero surface

### PR DESCRIPTION
**Pull request by pressel**
This PR incorporates into mam4xx a struct that has been added to haero for the purpose of moving information about the surface between the host model and mam4xx. This structure has been incorporated into the compute_tendencies and validate methods included in the `AeroProcess` class.

### CODE REVIEW

File: build-haero.sh [hunk 1]
```diff
@@ -73,6 +73,7 @@ fi
 echo "Cloning Haero repository into $(pwd)/.haero..."
 git clone git@github.com:eagles-project/haero.git .haero || exit
 cd .haero || exit
+git checkout -t origin/kylepres/surface.hpp
```
Comment (jeff-cohere) [inline build-haero.sh:76]: Great. We should delete this line after Haero gets merged (and add `[rebuild-haero]` to that commit)

**Review by odiazib — APPROVED** [review]

**Comment by jeff-cohere** [comment]
The PNNL CI thing will likely fail because it's not building against the right Haero, but once we've merged that Haero PR, we'll be able to tell it to do so.

**Review by jeff-cohere — APPROVED** [review]

**Review by mjs271 — APPROVED** [review]
Looks good--thanks for putting in the legwork on this!

**Comment by jeff-cohere** [comment]
Okay, the PNNL CI system hit an unrelated snag, so I'm merging this so we don't have conflicts with our new Haero and mam4xx.

**Pull request merged**